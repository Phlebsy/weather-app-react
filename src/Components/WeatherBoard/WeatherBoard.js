import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { convertFromKelvin, getIcon } from '../../utilities/utilities';

// WeatherBoard component represents the current clicked on forecast
// No logic, only presentational markup
export default class WeatherBoard extends Component {
  render() {
    const { forecast } = this.props;
    let { main } = forecast;

    return (
      <div className="weatherboard container px-0">
        <div className="d-flex justify-content-center">
          <div className="d-flex align-items-center justify-content-center">
            <div
              style={{ flexDirection: 'column' }}
              className="d-flex mx-4 align-items-center justify-content-center"
            >
              <i
                style={{ fontSize: '5rem' }}
                className={
                  'my-3 fas fa-' +
                  getIcon(forecast['dateTime'], forecast['weather'][0])
                }
              />
              <p className="text-capitalize">
                {forecast['weather'][0]['description']}
              </p>
            </div>
          </div>
          <div className="mx-4">
            <p className="text-danger text-center">
              {convertFromKelvin(main['temp_max'])}&deg;
            </p>
            <p style={{ fontSize: '2rem' }} className="text-center">
              {convertFromKelvin(main['temp'])}&deg;F
            </p>
            <p className="text-primary text-center">
              {convertFromKelvin(main['temp_min'])}&deg;
            </p>
          </div>
        </div>
        <div className="d-flex flex-columns justify-content-center">
          <div className="mx-3">
            {forecast['wind'['speed']]
              ? 0
              : forecast['wind']['speed'].toFixed(1)}{' '}
            MPH Wind
          </div>
          <div className="mx-3">{main.humidity}% Humidity</div>
        </div>
      </div>
    );
  }
}

WeatherBoard.propTypes = {
  forecast: PropTypes.object.isRequired
};
