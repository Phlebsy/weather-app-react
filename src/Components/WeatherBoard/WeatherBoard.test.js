import React from 'react';
import ReactDOM from 'react-dom';
import WeatherBoard from './WeatherBoard';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const forecastObj = {
  dateTime: '6:15',
  weather: [
    {
      description: 'clear sky',
      main: 'Clear'
    }
  ],
  main: {
    temp: 276.68,
    pressure: 1010,
    humidity: 60,
    temp_min: 275.95,
    temp_max: 277.55
  },
  wind: {
    speed: 4.1,
    deg: 320
  }
};

describe('<WeatherBoard />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<WeatherBoard forecast={forecastObj} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('accepts forecast prop', () => {
    const wrapper = shallow(<WeatherBoard forecast={forecastObj} />);
    expect(wrapper.instance().props.forecast.dateTime).toBe('6:15');
  });
  it('renders icon correctly', () => {
    const wrapper = shallow(<WeatherBoard forecast={forecastObj} />);
    expect(wrapper.find('.fa-sun').length).toBe(1);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<WeatherBoard forecast={forecastObj} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
