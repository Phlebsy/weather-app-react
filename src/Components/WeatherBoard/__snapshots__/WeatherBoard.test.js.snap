// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherBoard /> matches the snapshot 1`] = `
<div
  className="weatherboard container px-0"
>
  <div
    className="d-flex justify-content-center"
  >
    <div
      className="d-flex align-items-center justify-content-center"
    >
      <div
        className="d-flex mx-4 align-items-center justify-content-center"
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <i
          className="my-3 fas fa-sun"
          style={
            Object {
              "fontSize": "5rem",
            }
          }
        />
        <p
          className="text-capitalize"
        >
          clear sky
        </p>
      </div>
    </div>
    <div
      className="mx-4"
    >
      <p
        className="text-danger text-center"
      >
        39
        °
      </p>
      <p
        className="text-center"
        style={
          Object {
            "fontSize": "2rem",
          }
        }
      >
        38
        °F
      </p>
      <p
        className="text-primary text-center"
      >
        37
        °
      </p>
    </div>
  </div>
  <div
    className="d-flex flex-columns justify-content-center"
  >
    <div
      className="mx-3"
    >
      4.1
       
      MPH Wind
    </div>
    <div
      className="mx-3"
    >
      60
      % Humidity
    </div>
  </div>
</div>
`;
